### PR TITLE
Fix bug when multiple tags are present with same scope

### PR DESCRIPTION
**Pull request by ggverma**
The original aim of this PR was to fix Issue: #445 but it was fixed by https://github.com/vmware/ansible-for-nsxt/pull/439.

Now this PR only makes cosmetic improvements

### CODE REVIEW

**Comment by salv-orlando** [comment]
It's not very clear to me how this will address the fact that not all tags are update when multiple tags with the same scope are specified in `add_tags`. I understand how this PR addresses any issue with tag removal, but not addition. 

Can you provide more details?

File: plugins/modules/nsxt_vm_tags.py [hunk 1]
```diff
@@ -263,17 +259,14 @@ def realize():
...
-            if tag:
-                persistent_tags += tag,
 
-        final_tags = persistent_tags
+        final_tags = [tag for tag in all_tags if tag is not None]
         final_tags_set = _get_tags_as_set(tags=final_tags)
         for tag in _read_tags_from_module_params(module.params, 'add_tags'):
-            if TagElement(tag) not in final_tags_set:
+            tag_element = TagElement(tag)
+            if tag_element not in final_tags_set:
```
Comment (salv-orlando) [inline plugins/modules/nsxt_vm_tags.py:267]: In general this kind of check should not be needed with sets, why are we doing it?

**Comment by ggverma** [comment]
> It's not very clear to me how this will address the fact that not all tags are update when multiple tags with the same scope are specified in `add_tags`. I understand how this PR addresses any issue with tag removal, but not addition.
> 
> Can you provide more details?

That is because the issue with addition of tags existed before merging this PR https://github.com/vmware/ansible-for-nsxt/pull/439.

**Review by salv-orlando — APPROVED** [review]
LGTM!

**Pull request merged**